app.rb

```diff
@@ -0,0 +1,158 @@
+      list_person
+    end
+  end
+
+  def list_student
+    if @students.empty?
+      puts 'No students found. Please add some students to the list first.'
+    else
+      @students.each_with_index do |student, index|
+        puts "#{index}) [#{student.class}] ID: #{student.id}, Name: #{student.name},
+        Age: #{student.age}, classroom: #{student.classroom}"
+      end
+    end
+    nil
+  end
+
+  def list_teacher
+    if @teachers.empty?
+      puts 'No teachers found. Please add some teachers to the list first.'
+    else
+      @teachers.each_with_index do |teacher, index|
+        puts "#{index}) [#{teacher.class}] ID: #{teacher.id}, Name: #{teacher.name},
+        Age: #{teacher.age}, specialization: #{teacher.specialization}"
+      end
+    end
+    nil
+  end
+
+  # ========================= Create a person method ==============================
+  def create_person
+    print 'Do you want to add a student (1) or a teacher (2)? [Input the number]: '
+    select_person = gets.chomp.to_i
+    case select_person
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts "\n\n******** Invalid input, please try again!!! ************ \n\n"
+      create_person
+    end
+  end
+
+  def create_student
+    print 'Classroom: '
+    student_classroom = gets.chomp
+    print 'Age: '
+    student_age = gets.chomp.to_i
+    print 'Name: '
+    student_name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp
+    student = Student.new(student_classroom, student_age, student_name, parent_permission: parent_permission)
+    @students << student
+    puts 'Student created successfully'
+  end
+
+  def create_teacher
+    print 'Age: '
+    teacher_age = gets.chomp.to_i
+    print 'Name: '
+    teacher_name = gets.chomp
+    print 'Specialization: '
+    teacher_specialization = gets.chomp
+    teacher = Teacher.new(teacher_specialization, teacher_age, teacher_name)
+    @teachers << teacher
+    puts 'Teacher created successfully'
+  end
+
+  # ========================= Create a book method =============================
+  def create_book
+    print 'Title: '
+    book_title = gets.chomp
+    print 'Author: '
+    book_author = gets.chomp
+    book = Book.new(book_title, book_author)
+    @books << book
+    puts 'Book created successfully'
+  end
+
+  # ========================= Create a rental method =============================
+  def create_rental
+    if @books.empty? || (@teachers.empty? && @students.empty?)
+      puts 'list are empty'
+    else
+      puts 'Select a book from the following list by number'
+      list_books
+      book_number = gets.chomp.to_i
+      book_name = @books[book_number]
+
+      puts 'Select a person from the following list by number (not id)'
+      list_people
+      person_number = gets.chomp.to_i
+      person_name = @teachers[person_number] || @students[person_number]
+
```

Comment:
- [ ] Great work done champ 👏. However, there is a little bug I noticed in the code. When I try to create a _**rental**_, I notice that the rental is always created for a teacher instance even if I choose a Student. The reason for this is because of this line of code `person_name = @teachers[person_number] || @students[person_number]`. 

With this line of code, if I have a `Teacher` instance with the listed number, it will always create a rental with the `Teacher` instance first even if I want the rental to be for a `Student` with the same number. It is only when I don't have a `Teacher` instance with that number that it will record for a `Student` instance. This does not make it specific. I would advise you either list all instances of people as seen in the video guide or have a way of getting the accurate instance and the number of the person I wish to attach the rental to. 👍 💻